runtime-decompiler/src/main/java/org/jrd/backend/data/Help.java

```diff
@@ -0,0 +1,244 @@
+
+        void printSavingOptionsSubheading();
+
+        default void printOptions() {
+            printMainOptionsSubheading();
+            printOptions(ALL_OPTIONS);
+
+            printSavingOptionsSubheading();
+            printOptions(SAVING_OPTIONS);
+        }
+
+        void printOptions(Map<String, String> map);
+
+        String optionize(String[] options);
+
+        default String indent() {
+            return "  ";
+        }
+
+        default String[] launchOptions() {
+            return new String[] {
+                    "# launches GUI",
+                    optionize(UNSAVABLE_OPTIONS),
+                    optionize(SAVABLE_OPTIONS) + savingModifiers()
+            };
+        }
+
+        String launcher();
+
+        String savingModifiers();
+
+        void printNotesHeading();
+
+        void printNotes();
+    }
+
+    private static class CliHelpFormatter implements HelpFormatter {
+
+        @Override
+        public void printTitle() { }
+
+        @Override
+        public void printName() { }
+
+        @Override
+        public void printUsageHeading() {
+            System.out.println("Usage:");
+        }
+
+        @Override
+        public void printOptionsHeading() { }
+
+        @Override
+        public void printMainOptionsSubheading() {
+            System.out.println("Available options:");
+        }
+
+        @Override
+        public void printSavingOptionsSubheading() {
+            System.out.println("Saving modifiers:");
+        }
+
+        @Override
+        public void printOptions(Map<String, String> map) {
+            for (Map.Entry<String, String> entry : map.entrySet()) {
+                String format = entry.getKey();
+                String initialSpacing = " ".repeat(LONGEST_FORMAT_LENGTH - format.length());
+                String interlineSpacing = "\n" + indent() + " ".repeat(LONGEST_FORMAT_LENGTH);
+
+                System.out.println(indent() + format + initialSpacing +
+                        entry.getValue().replaceAll("\n", interlineSpacing));
+            }
+        }
+
+        @Override
+        public String optionize(String[] options) {
+            return "(" + String.join("|", options) + ")";
+        }
+
+        @Override
+        public String savingModifiers() {
+            return " [" + SAVEAS + " [" + SAVELIKE + "]]";
+        }
+
+        @Override
+        public String launcher() {
+            return (isOsWindows() ? LAUNCHER_WINDOWS : LAUNCHER_LINUX) + " [" + VERBOSE + "] ";
+        }
+
+        @Override
+        public void printNotesHeading() {
+            System.out.println("Additional information");
+        }
+
+        @Override
+        public void printNotes() {
+            for (Map.Entry<String, String[]> entry : NOTES.entrySet()) {
+                System.out.println(indent() + entry.getKey());
+                for (String item : entry.getValue()) {
+                    System.out.println(indent() + indent() + "- " + item);
```

Comment:
maybe indent can take parameter number and inent n times of base unit?

Reply:
smart!